account-kit/core/src/actions/createAccount.ts

```diff
@@ -94,18 +97,26 @@ export async function createAccount<TAccount extends SupportedAccountTypes>(
     throw new Error("Signer not connected");
   }
 
-  const cachedAccount = accounts[chain.id]?.[type];
+  const cachedAccount = (
+    type === "ModularAccountV2"
+      ? accounts[chain.id]?.["ModularAccountV2"]?.[mode]
+      : accounts[chain.id]?.[type]
+  ) as AccountState<TAccount>;
```

Comment:
Is line 106 safe? We're casting it to a non-nullable type, but it seems like the actual thing can be undefined?

Reply:
i think it is always initialized with a `defaultAccountState` like `{
  status: "DISCONNECTED",
  account: undefined,
}`